### PR TITLE
IGNITE-13566

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/rules/OrToUnionRuleTest.java [hunk 1]
```diff
@@ -224,7 +238,7 @@ public void testNonIndexedOrToUnionAllRewrite() throws Exception {
      * @throws Exception If failed.
      */
     @Test
-    @Ignore("https://issues.apache.org/jira/browse/IGNITE-12819")
+    //@Ignore("https://issues.apache.org/jira/browse/IGNITE-12819")
```
Comment (Berkof) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/rules/OrToUnionRuleTest.java:241]: Commented code

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java [hunk 2]
```diff
@@ -2871,6 +2871,10 @@ private void checkxEnabled() throws IgniteException {
         });
     }
 
+    public boolean useExperimentalEngine() {
```
Comment (Berkof) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java:2874]: javadoc

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java [hunk 3]
```diff
@@ -132,15 +132,15 @@ public String indexName() {
...
+                double selectivity0 = mq.getSelectivity(this, RexUtil.composeConjunction(builder, lowerCondition()));
 
                 selectivity -= 1 - selectivity0;
 
                 cost += Math.log(rows);
             }
 
-            if (upperCondition() != null) {
-                double selectivity0 = mq.getSelectivity(this, RexUtil.composeDisjunction(builder, upperCondition()));
+            if (upperCondition() != null && lowerCondition() != null && !lowerCondition().equals(upperCondition())) {
```
Comment (korlov42) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java:142]: > !lowerCondition().equals(upperCondition())
this doesn't work with condition, only with bounds. I think it's be better to leave it as is for now and fix it under https://issues.apache.org/jira/browse/IGNITE-14034

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/SqlFieldsQueryUsageTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    /**
+     * Temporary redirects create|drop|alter commands into h2 engine.
+     */
+    @Test
+    public void testUseH2Functionality() {
+        execute(grid(0), "CREATE TABLE IF NOT EXISTS Person(\"id\" INT, PRIMARY KEY(\"id\"), \"name\" VARCHAR)");
+
+        execute(grid(0), "alter table Person add column age int");
+        execute(grid(0),"drop table Person");
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/SqlFieldsQueryUsageTest.java:64]:
```suggestion
        execute(grid(0), "drop table Person");
```

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/SqlFieldsQueryUsageTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                .setQueryEntities(singletonList(projEntity))
+                .setSqlSchema("PUBLIC");
+
+        IgniteCache<Integer, Integer> devCache = grid(0).createCache(projCfg);
+
+        assertFalse(grid(0).configuration().isClientMode());
+
+        devCache.put(1, 2);
+
+        assertEquals(1, execute(client,"SELECT * FROM Developer").size());;
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/SqlFieldsQueryUsageTest.java:91]:
```suggestion
        assertEquals(1, execute(client, "SELECT * FROM Developer").size());;
```

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java (hunk 3 above)
Comment (zstan) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java:142]: discussed in private, i don`t touch this place.

**Comment by dspavlov** [comment]
This PR is associated with a resolved ticket. Closing the PR, please reopen it if you still need it.

**Pull request closed without merging**